### ISSUE TITLE
Speed up Qiskit

### ISSUE THREAD

**Issue by ajavadia** [qiskit-community/qiskit-camp-europe-19#30]
# Abstract
This is a broad project for benchmarking, profiling, and speeding up Qiskit. I can help with a representative set of benchmarks to try this out on. The bottlenecks could be transpiler passes, object serialization and validation, simulation. The first step would be to do a detailed profiling on each of these benchmarks and understand the sources of slowdown.
The second step is to speed up those bottlenecks. My guesstimate is that a 20-30% speedup should be attainable.
 
# Description 

# Members

- @ruimaranhao 
- @amarchenkova
- @picardch
- @danielfobooss
- IBM Coach: @diego-plan9 @1ucian0 @hhorii 

# Deliverable
A detailed profile of runtime for each benchmark, and an analysis of what is slow and why.
Some faster alternative implementations.

# GitHub repo
https://github.com/Qiskit/qiskit-aqua/pull/684/

**Comment by amarchenkova** [comment]
I’m interested in this!

**Comment by picardch** [comment]
I am also interested in this project.

**Comment by ruimaranhao** [comment]
Cool one. Also interested.

**Comment by danielfobooss** [comment]
Interested!

**Comment by ruimaranhao** [comment]
I would say that some of the issues in #1 will be fixed during this one too.

**Comment by ruimaranhao** [comment]
@ajavadia can you share the representative benchmarks ?

**Comment by ajavadia** [comment]
yes see here: https://github.com/Qiskit/qiskit/tree/master/test/benchmarks

I'll stop by in a bit.
